### ISSUE TITLE
Resource deployment azurerm_app_service_certificate_binding does not updates the state file when ssl_state = "IpBasedEnabled" in the first apply operation

### ISSUE THREAD

**Issue by kovaliovsg** [hashicorp/terraform-provider-azurerm#11854]
### Description

This is a typical scenario for binding an apex domain (root domain Record A) to a web application.
When we bind a certificate to the apex domain using ssl_state = "IpBasedEnabled", a new static IP address is generated for the web application that should be used to create Record A in the DNS Zone to allow accessing the Web Application using the apex domain with SSL (https://contoso.com).

When step 3 is executed, the resource azurerm_app_service_custom_hostname_binding.apex_domain_binding is physically updated in azure with the new attributes:
      + ssl_state           = "IpBasedEnabled"
      + thumbprint      = "SomeThumbprint"
      + virtual_ip          = "SomeIPAddress"

However, the state file is not refreshed with these new attributes after the step 3, so the step 4 fails because the interpolation records = [azurerm_app_service_custom_hostname_binding.apex_domain_binding.virtual_ip] is not going to work because the state file does not contain virtual_ip attribute.

Only when I run the terraform apply the second time, the state file will be refreshed with the virtual_ip and the interpolation will succeed.

### Terraform (and AzureRM Provider) Version

Terraform v0.15.4
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.60.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.1

### Affected Resource(s)

azurerm_app_service_certificate_binding
azurerm_app_service_custom_hostname_binding

### Terraform Configuration Files

```
// Step1 Validate Web Application by adding a TXT record

resource "azurerm_dns_txt_record" "apex_domain_txt_record" {
  provider                       = azurerm.prod
  name                           = "asuid"
  zone_name                  = var.webapp_zone_name
  resource_group_name = var.webapp_zone_resource_group_name
  ttl                                 = 300
  record {
    value = azurerm_app_service.app_service.custom_domain_verification_id
  }
}

// Step2 Add apex custom domain to the web application. For example contoso.com

resource "azurerm_app_service_custom_hostname_binding" "apex_domain_binding" {
  hostname                    = var.webapp_zone_name
  app_service_name       = azurerm_app_service.app_service.name
  resource_group_name = var.webapp_resource_group_name
  depends_on                = [azurerm_dns_txt_record.apex_domain_txt_record]
}

// Step3 create certificate binding using ssl_state = "IpBasedEnabled"

resource "azurerm_app_service_certificate_binding" "apex_certificate_binding" {
  hostname_binding_id = azurerm_app_service_custom_hostname_binding.apex_domain_binding.id
  certificate_id               = azurerm_app_service_certificate.app_service_certificate.id
  ssl_state                      = "IpBasedEnabled"
}

// Step4 Create A record in DNS zone using a new static public IP address produced in step 3

resource "azurerm_dns_a_record" "a_record" {
  name                           = "@"
  zone_name                  = var.webapp_zone_name
  resource_group_name = var.webapp_zone_resource_group_name
  ttl                                 = 300
  records                        = [azurerm_app_service_custom_hostname_binding.apex_domain_binding.virtual_ip]
  depends_on                = [azurerm_app_service_certificate_binding.apex_certificate_binding]
}
```
### Expected Behaviour

The state file is refreshed for resource azurerm_app_service_custom_hostname_binding.apex_domain_binding right after step 3 so the virtual_ip attribute can be fetched by the resource azurerm_dns_a_record.a_record in step 4 in the first apply operation.

### Actual Behaviour

The state file is not refreshed for resource azurerm_app_service_custom_hostname_binding.apex_domain_binding with the new attributes produced in step 3.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
